finagle-natchez/src/main/scala/com/dwolla/util/async/finagle/TracedThriftServer.scala

```diff
@@ -44,39 +43,35 @@ object TracedThriftServer {
                                                                                             )
                                                                                             (implicit ec: ExecutionContext,
                                                                                              LocalSpan: Local[F, Span[F]]): Resource[F, ListeningServer] =
+    TracedThriftServer(addr, label, iface, entryPoint, Span.Options.Defaults.withSpanKind(Span.SpanKind.Server))
```

Comment:
Adding `Span.Options.Defaults.withSpanKind(Span.SpanKind.Server)` is the key change in this PR; the rest is really just refactoring.